packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkStateManager.ts

```diff
@@ -371,6 +374,69 @@ export class ForkStateManager implements StateManager {
     return value;
   }
 
+  // the following methods are copied verbatim from
+  // DefaultStateManager
+
+  public isWarmedAddress(address: Buffer): boolean {
```

Comment:
What about doing `return DefaultStateManager.prototype.isWarmedAddress.call(this, address)` instead? The same for the rest of the methods.

Reply:
Good idea, I'll give it a try.

Reply:
Done. Notice that this adds an implicit dependency between those method calls and certain instance variables that have to exist (`_accessedStorage` and `_accessedStorageReverted`). `tsc` doesn't seem to catch those errors.

Reply:
Maybe leave a comment? Either thing is dangerous in its own way

Reply:
There is one:

```typescript
  // used by the DefaultStateManager calls
  private _accessedStorage: Array<Map<string, Set<string>>> = [new Map()];
  private _accessedStorageReverted: Array<Map<string, Set<string>>> = [
    new Map(),
  ];
```